### PR TITLE
[DF][NFC] Add missing documentation to several RDF classes. 

### PR DESCRIPTION
**Pull request by martamaja10**
Several of the RDF classes (RMetaData, RSample, RSampleInfo) were missing some of the documentation and examples of usage. Additionally, in the process of fixing those noticed a few typos in RDatasetSpec and df106 tutorials.

### CODE REVIEW

File: tree/dataframe/inc/ROOT/RDF/RMetaData.hxx [hunk 1]
```diff
@@ -32,10 +32,20 @@ namespace Experimental {
...
- This class should be passed to an RSample object which represents a single dataset sample.
- Once a dataframe is built with RMetaData object, it could be accessed via DefinePerSample.
+ An example of creating the RMetaData object:
+ ~~~{.cpp}
+ ROOT::RDF::Experimental::RMetaData meta;
+ meta.Add("sample_name", "name"");
+ meta.Add("luminosity", 10064);
+ meta.Add("xsecs", 1.0);
+ ~~~
+
```
Comment (vepadulano) [inline tree/dataframe/inc/ROOT/RDF/RMetaData.hxx:44]: Maybe we could also add a comment here specifying that the only supported types in RMetaData are `std::string`,`int`,`double`

**Review by vepadulano — APPROVED** [review]
Thanks a lot @martamaja10 ! A few minor suggestions

File: tree/dataframe/inc/ROOT/RDF/RSample.hxx [hunk 2]
```diff
@@ -40,6 +51,7 @@ class RSample {
     * They can contain the globbing characters supported by TChain. See TChain::Add for more information.
     */
    std::vector<std::string> fFileNameGlobs;
+   /// An instance of the RMetaData class.
    RMetaData fMetaData;
 
    unsigned int fSampleId{0}; // global sample index, set inside of the RDatasetSpec
```
Comment (enirolf) [inline tree/dataframe/inc/ROOT/RDF/RSample.hxx:57]: To make this one also show up in Doxygen and make it consistent with the other docstrings:
```suggestion
   /// global sample index, set inside of the RDatasetSpec
   unsigned int fSampleId{0};
```

File: tree/dataframe/inc/ROOT/RDF/RSampleInfo.hxx [hunk 3]
```diff
@@ -25,7 +25,7 @@ namespace ROOT {
 namespace RDF {
 
 /// This type represents a sample identifier, to be used in conjunction with RDataFrame features such as
-/// DefinePerSample() and per-sample callbacks.
+/// \ref ROOT::RDF::RInterface< Proxied, DS_t >::DefinePerSample "DefinePerSample()" and per-sample callbacks.
```
Comment (enirolf) [inline tree/dataframe/inc/ROOT/RDF/RSampleInfo.hxx:28]: Not a suggestion but I wonder why this doesn't automatically gets turned into a reference but `AsString()` in the line below does 🤔

File: tree/dataframe/src/RMetaData.cxx [hunk 4]
```diff
@@ -36,77 +36,109 @@ RMetaData &RMetaData::operator=(RMetaData &&) = default;
 
 RMetaData::~RMetaData() = default;
 
+/// @brief Add an RMetaData class instance.
+/// @param key input key for a given RMetaData instance.
```
Comment (enirolf) [inline tree/dataframe/src/RMetaData.cxx:40]: This is very nitpicky, but:
```suggestion
/// @param[in] key input key for a given RMetaData instance.
```
(taken from https://root.cern/for_developers/doxygen/#method-documentation, also applies to the subsequent docstrings)

**Review by enirolf — COMMENTED** [review]
Thanks! Also a few small suggestions from my side

File: tree/dataframe/inc/ROOT/RDF/RSampleInfo.hxx (hunk 3 above)
Comment (martamaja10) [reply tree/dataframe/inc/ROOT/RDF/RSampleInfo.hxx:28]: AsString() is a part of this class so I think that's why you don't need anything extra

File: tree/dataframe/src/RMetaData.cxx (hunk 4 above)
Comment (martamaja10) [reply tree/dataframe/src/RMetaData.cxx:40]: thanks! I missed it, definitely, we should be consistent

**Pull request merged**